Client/Frontend/Home/ActivityStreamPanel.swift

```diff
@@ -343,17 +334,54 @@ extension ActivityStreamPanel {
     }
 
     @objc private func longPress(longPressGestureRecognizer: UILongPressGestureRecognizer) {
-        if longPressGestureRecognizer.state == UIGestureRecognizerState.Began {
-            let touchPoint = longPressGestureRecognizer.locationInView(self.view)
-            if let indexPath = tableView.indexPathForRowAtPoint(touchPoint) {
-                if Section(indexPath.section) == .Highlights {
-                    presentContextMenu(history[indexPath.row], indexPath: indexPath)
-                }
+        guard longPressGestureRecognizer.state == UIGestureRecognizerState.Began else { return }
+
+        let touchPoint = longPressGestureRecognizer.locationInView(self.view)
+        guard let indexPath = tableView.indexPathForRowAtPoint(touchPoint) else { return }
+
+        let section = Section(indexPath.section)
+        if section == .Highlights {
+            guard let highlightCell = tableView.cellForRowAtIndexPath(indexPath) as! SimpleHighlightCell? else { return }
+            let headerIconHandler = { return highlightCell.siteImageView.image }
+            let headerIconBackgroundHandler = { return highlightCell.siteImageView.backgroundColor }
+            presentContextMenu(history[indexPath.row], section: section, indexPath: indexPath, headerIconHandler: headerIconHandler, headerIconBackgroundHandler: headerIconBackgroundHandler)
+        } else {
+            let topSiteCell = self.tableView.cellForRowAtIndexPath(indexPath) as! ASHorizontalScrollCell
+            let touchPointWithinTopSiteCell = longPressGestureRecognizer.locationInView(topSiteCell.collectionView)
+            if let indexPath = topSiteCell.collectionView.indexPathForItemAtPoint(touchPointWithinTopSiteCell) {
+                let topSiteItemCell = topSiteCell.collectionView.cellForItemAtIndexPath(indexPath) as! TopSiteItemCell
+                let headerIconHandler = { return topSiteItemCell.imageView.image }
```

Comment:
Thats a problem in the tableviewcell. Selecting it shouldn't cause the background color to disappear. Still using closures here is not a good idea.